internal/commands/user/list.go

```diff
@@ -80,30 +80,30 @@ func (cmd *CommandList) Handler(profile *cli.Profile, ui terminal.UI) error {
 	return nil
 }
 
-const (
-	userDataEmail = "email"
-	userDataName  = "name"
-)
-
 // Feedback is the command feedback
 func (cmd *CommandList) Feedback(profile *cli.Profile, ui terminal.UI) error {
 	if len(cmd.users) == 0 {
 		return ui.Print(terminal.NewTextLog("No available users to show"))
 	}
 
-	var usersByProviderType = make(map[string][]realm.User)
+	usersByProviderType := map[realm.ProviderType][]realm.User{}
 	for _, user := range cmd.users {
 		for _, identity := range user.Identities {
 			usersByProviderType[identity.ProviderType] = append(usersByProviderType[identity.ProviderType], user)
 		}
 	}
 
 	var logs []terminal.Log
-	for providerType, users := range usersByProviderType {
+	for _, pt := range realm.ValidProviderTypes {
+		providerType := realm.ProviderType(pt)
```

Comment:
[question] why this change?  couldn't you keep just iterating over `usersByProviderType` here?

Reply:
Maps, do not guarantee ordering. Was getting the occasional test failure.

Reply:
If we don't care about ordering, I could look to see about making the test work for any type of ordering

Reply:
ah completely forgot we were relying on deterministic output for the tests.  i like the purpose behind this change then, although I'd prefer we didnt do this `providerType := realm.ProviderType(pt)` on the first line of each loop.

what would it take to keep `realm.ValidProviderTypes` to be typed as `realm.AuthProvderTypes`?  a brief look on my end seems to indicate its possible, but just might mean we have to implement a `func (types AuthProviderTypes) Strings() []string` method, or adjust the `flags.EnumSetValue` to hold onto `validValues` as a `[]interface{}` instead of string array.  thoughts?